pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -429,103 +523,24 @@ def create_meas_results_per_qb(self):
-            self.cal_states_rotations = self.get_param_value(
-                'cal_states_rotations', default_value=cal_states_rots)
-            sweep_points_w_calpts = \
-                {qbn: {'sweep_points': self.cp.extend_sweep_points(
-                    self.proc_data_dict['sweep_points_dict'][qbn][
-                        'sweep_points'], qbn)} for qbn in self.qb_names}
-            self.proc_data_dict['sweep_points_dict'] = sweep_points_w_calpts
-        except TypeError as e:
-            log.error(e)
-            log.warning("Failed retrieving cal point objects or states. "
-                        "Please update measurement to provide cal point object "
-                        "in metadata. Trying to get them using the old way ...")
-            self.cal_states_rotations = self.get_param_value(
-                'cal_states_rotations', default_value=cal_states_rotations) \
-                if self.rotate else cal_states_rotations
-            self.cal_states_dict = self.get_param_value('cal_states_dict',
-                                                         default_value={})
-
-        if self.get_param_value('global_PCA') is not None:
-            log.warning('Parameter "global_PCA" is deprecated. Please set '
-                        'rotation_type="global_PCA" instead.')
-        self.rotation_type = self.get_param_value(
-            'rotation_type',
-            default_value='cal_states' if self.rotate else 'no_rotation')
 
         # create projected_data_dict
-        self.data_to_fit = deepcopy(self.get_param_value('data_to_fit'))
-        if self.data_to_fit is None:
-            # If we have cal points, but data_to_fit is not specified,
-            # choose a reasonable default value. In cases with only two cal
-            # points, this decides which projected plot is generated. (In
-            # cases with three cal points, we will anyways get all three
-            # projected plots.)
-            if 'e' in self.cal_states_dict.keys():
-                self.data_to_fit = {qbn: 'pe' for qbn in self.qb_names}
-            elif 'g' in self.cal_states_dict.keys():
-                self.data_to_fit = {qbn: 'pg' for qbn in self.qb_names}
-            else:
-                self.data_to_fit = {}
-
-
-        # TODO: Steph 15.09.2020
-        # This is a hack to allow list inside data_to_fit. These lists are
-        # currently only supported by MultiCZgate_CalibAnalysis
-        for qbn in self.data_to_fit:
-            if isinstance(self.data_to_fit[qbn], (list, tuple)):
-                self.data_to_fit[qbn] = self.data_to_fit[qbn][0]
         if self.rotate or self.rotation_type == 'global_PCA':
             self.cal_states_analysis()
         else:
             # this assumes data obtained with classifier detector!
             # ie pg, pe, pf are expected to be in the value_names
+            if not len(self.data_to_fit):
+                self.data_to_fit = {qbn: 'pe' for qbn in self.qb_names}
```

Comment:
Why is this a reasonable data_to_fit not yet ensured at this point? E.g. in the part "# get data_to_fit" in line 318?

If it is really needed to do this late point, please add a comment that explains why.

Reply:
Yes, can be removed now.